src/core/init/directfile_content_initializer.ts

```diff
@@ -266,28 +266,38 @@ function getDirectFileInitialTime(
   }
 
   const duration = mediaElement.duration;
-
   if (typeof startAt.fromLastPosition === "number") {
-    if (isNullOrUndefined(duration) || !isFinite(duration)) {
-      log.warn(
-        "startAt.fromLastPosition set but no known duration, " + "beginning at 0.",
-      );
-      return 0;
+    if (!isNullOrUndefined(duration) && isFinite(duration)) {
+      return Math.max(0, duration + startAt.fromLastPosition);
     }
-    return Math.max(0, duration + startAt.fromLastPosition);
+
+    if (mediaElement.seekable.length > 0) {
+      const lastSegmentEnd = mediaElement.seekable.end(mediaElement.seekable.length - 1);
+      if (isFinite(lastSegmentEnd)) {
+        return Math.max(0, lastSegmentEnd + startAt.fromLastPosition);
+      }
+    }
+    log.warn(
+      "Init: startAt.fromLastPosition set but no known duration, " +
+        "it may be too soon to seek",
+    );
+    return undefined;
   } else if (typeof startAt.fromLivePosition === "number") {
     const livePosition =
       mediaElement.seekable.length > 0 ? mediaElement.seekable.end(0) : duration;
     if (isNullOrUndefined(livePosition)) {
       log.warn(
-        "startAt.fromLivePosition set but no known live position, " + "beginning at 0.",
+        "Init: startAt.fromLivePosition set but no known live position, " +
+          "beginning at 0.",
       );
       return 0;
     }
     return Math.max(0, livePosition + startAt.fromLivePosition);
   } else if (startAt.percentage != null) {
     if (isNullOrUndefined(duration) || !isFinite(duration)) {
-      log.warn("startAt.percentage set but no known duration, " + "beginning at 0.");
+      log.warn(
+        "Init: startAt.percentage set but no known duration, " + "beginning at 0.",
```

Comment:
And why is this one changed but not like the other one?

Reply:
It fits the 90 col from the prettier config so it stay on one line.
This has changed because I added `"Init:"` prefix to the log